File: 1.js
// Day 1
// http://adventofcode.com/day/1

function part1(){
	var input = document.getElementsByTagName("pre")[0].innerHTML;
	var floor = 0;

	for(var i = 0; i < input.length; i++){
		input.charAt(i) == "(" ? floor++ : floor--;
	}
	return floor;
}

function part2(){
	var input = document.getElementsByTagName("pre")[0].innerHTML;
	var floor = 0;

	for(var i = 0; i < input.length; i++){
		input.charAt(i) == "(" ? floor++ : floor--;
		if(floor < 0){
			return (i + 1);
		}
	}
}

File: 1.py
#!/usr/bin/env python3
'''
    Day 1
    http://adventofcode.com/2015/day/1
'''

from utils import get_input_filename

DAY_NUM = 1

def part1():
    level = 0
    with open(get_input_filename(DAY_NUM)) as f:
        for line in f:
            for c in line:
                if c == '(':
                    level += 1
                elif c == ')':
                    level -= 1
    return level

def part2():
    level = 0
    position = 0
    with open(get_input_filename(DAY_NUM)) as f:
        for line in f:
            for c in line:
                position += 1
                if c == '(':
                    level += 1
                elif c == ')':
                    level -= 1
                if level < 0:
                    return position


if __name__ == '__main__':
    print(part1())
    print(part2())
